connection/cluster/ClusterClient.java

```diff
@@ -315,12 +314,14 @@ private ClusterDatabase fetchDatabaseReplicas() {
                     if (!isNonExistentRaftClusterInfo(clusterDatabase)) {
                         clusterDatabases.put(database, clusterDatabase);
                     }
+                    roundRobinIndex = (roundRobinIndex + i + 1) % clusterAddresses.size();
                     return clusterDatabase;
                 } catch (TypeDBClientException e) {
                     if (UNABLE_TO_CONNECT.equals(e.getErrorMessage())) {
                         LOG.debug("Failed to fetch replica info for database '" + database + "' from " +
                                 serverAddress + ". Attempting next server.", e);
                     } else {
+                        roundRobinIndex = (roundRobinIndex + i + 1) % clusterAddresses.size();
```

Comment:
This would arguably be simpler if we had a `finally { roundRobinIndex = (roundRobinIndex + 1) % clusterAddresses.size(); }`. Then we can simply `clusterAddresses.get(roundRobinIndex);` at the start of each iteration.

Reply:
Fair. Doesn't this suffer from the "Querying node 1 twice in a row, if node 0 is unreachable" problem?

Reply:
Updated.
I think I shied away from it initially  because it had two variables being updated independently in a loop.